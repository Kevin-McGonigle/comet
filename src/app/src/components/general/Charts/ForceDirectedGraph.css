.forceDirectedGraph {
    max-width: 400px;
    background-color: black;
}

.title {
    color: #13b3c2;
    margin: 15px;
    font-family: 'Gill Sans', "Roboto", "Lucida Grande", Verdana, Arial, sans-serif;
    font-size: 11px;
    line-height: 14px;
    text-transform: uppercase;
    letter-spacing: 1px;
    font-weight: bold;
    position: absolute;
}

.test {
    width: 70px;
    color: black;
}