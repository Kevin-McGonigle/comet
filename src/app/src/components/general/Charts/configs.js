import React from "react";
import ClassDiagram from "./ClassDiagram/ClassDiagram";
import { DependencyGraphNodeContainer } from './DependencyGraph/DependencyGraphContainer'

export const staticConfig = {
    "automaticRearrangeAfterDropNode": false,
    "collapsible": false,
    "directed": true,
    "focusAnimationDuration": 0.75,
    "focusZoom": 1,
    "highlightDegree": 1,
    "highlightOpacity": 1,
    "linkHighlightBehavior": false,
    "maxZoom": 8,
    "minZoom": 1,
    "nodeHighlightBehavior": true,
    "staticGraphWithDragAndDrop": true,
    "panAndZoom": true,
    "d3": {
        "alphaTarget": 0.05,
        "gravity": -100,
        "linkLength": 100,
        "linkStrength": 1,
        "disableLinkForce": false
    },
    "node": {
        "color": "#3498db",
        "fontColor": "#1e272e",
        "fontSize": 20,
        "fontWeight": "normal",
        "highlightColor": "SAME",
        "highlightFontSize": 8,
        "highlightFontWeight": "normal",
        "highlightStrokeColor": "SAME",
        "highlightStrokeWidth": "SAME",
        "labelProperty": "id",
        "mouseCursor": "pointer",
        "opacity": 1,
        "renderLabel": true,
        "size": 200,
        "strokeColor": "#3498db",
        "strokeWidth": 1.5,
        "svg": "",
        "symbolType": "square"
    },
    "link": {
        "color": "#2c3e50",
        "fontColor": "black",
        "fontSize": 8,
        "fontWeight": "normal",
        "highlightColor": "true",
        "highlightFontSize": 8,
        "highlightFontWeight": "normal",
        "labelProperty": "label",
        "mouseCursor": "pointer",
        "opacity": 1,
        "renderLabel": false,
        "semanticStrokeWidth": false,
        "strokeWidth": 1.5,
        "markerHeight": 6,
        "markerWidth": 6
    }   
};

export const dynamicConfig = {
    "automaticRearrangeAfterDropNode": true,
    "collapsible": false,
    "directed": true,
    "focusAnimationDuration": 0.75,
    "focusZoom": 1,
    "height": 400,
    "highlightDegree": 1,
    "highlightOpacity": 1,
    "linkHighlightBehavior": false,
    "maxZoom": 8,
    "minZoom": 1,
    "nodeHighlightBehavior": false,
    "panAndZoom": false,
    "staticGraph": false,
    "staticGraphWithDragAndDrop": false,
    "width": 800,
    "d3": {
      "alphaTarget": 0.05,
      "gravity": -100,
      "linkLength": 100,
      "linkStrength": 1,
      "disableLinkForce": false
    },
    "node": {
      "color": "#3498db",
      "fontColor": "black",
      "fontSize": 8,
      "fontWeight": "normal",
      "highlightColor": "SAME",
      "highlightFontSize": 8,
      "highlightFontWeight": "normal",
      "highlightStrokeColor": "SAME",
      "highlightStrokeWidth": "SAME",
      "labelProperty": "id",
      "mouseCursor": "pointer",
      "opacity": 1,
      "renderLabel": true,
      "size": 200,
      "strokeColor": "#3498db",
      "strokeWidth": 1.5,
      "svg": "",
      "symbolType": "circle"
    },
    "link": {
      "color": "#2c3e50",
      "fontColor": "black",
      "fontSize": 8,
      "fontWeight": "normal",
      "highlightColor": "true",
      "highlightFontSize": 8,
      "highlightFontWeight": "normal",
      "labelProperty": "label",
      "mouseCursor": "pointer",
      "opacity": 1,
      "renderLabel": false,
      "semanticStrokeWidth": false,
      "strokeWidth": 1.5,
      "markerHeight": 6,
      "markerWidth": 6,
      "type": "CURVE_SMOOTH"
    }
}

export const classDiagramConfig = {
    "automaticRearrangeAfterDropNode": true,
    "collapsible": false,
    "directed": true,
    "focusAnimationDuration": 0.75,
    "focusZoom": 1,
    "highlightDegree": 1,
    "highlightOpacity": 1,
    "linkHighlightBehavior": false,
    "maxZoom": 2,
    "minZoom": 1,
    "nodeHighlightBehavior": true,
    "panAndZoom": true,
    "d3": {
        "alphaTarget": 0.05,
        "gravity": -100,
        "linkLength": 100,
        "linkStrength": 1,
        "disableLinkForce": false
    },
    "node": {
        "color": "#3498db",
        "fontColor": "#1e272e",
        "fontWeight": "normal",
        "highlightColor": "SAME",
        "highlightFontSize": 8,
        "highlightFontWeight": "normal",
        "highlightStrokeColor": "SAME",
        "highlightStrokeWidth": "SAME",
        "labelProperty": "id",
        "mouseCursor": "pointer",
        "opacity": 1,
        "renderLabel": true,
        "size": 200,
        "strokeColor": "#3498db",
        "strokeWidth": 1.5,
        "svg": "",
        "symbolType": "square",
        viewGenerator: node => <ClassDiagram data={node} />,

    },
    "link": {
        "color": "#2c3e50",
        "fontColor": "black",
        "fontSize": 8,
        "fontWeight": "normal",
        "highlightColor": "true",
        "highlightFontSize": 8,
        "highlightFontWeight": "normal",
        "labelProperty": "label",
        "mouseCursor": "pointer",
        "opacity": 1,
        "renderLabel": true,
        "semanticStrokeWidth": false,
        "strokeWidth": 1.5,
        "markerHeight": 6,
        "markerWidth": 6
    }   
}

export const dependencyGraphConfig = {
    "automaticRearrangeAfterDropNode": true,
    "collapsible": false,
    "directed": true,
    "focusAnimationDuration": 0.75,
    "focusZoom": 1,
    "highlightDegree": 1,
    "highlightOpacity": 1,
    "linkHighlightBehavior": false,
    "maxZoom": 2,
    "minZoom": 1,
    "nodeHighlightBehavior": true,
    "panAndZoom": true,
    "d3": {
        "alphaTarget": 0.05,
        "gravity": -100,
        "linkLength": 100,
        "linkStrength": 1,
        "disableLinkForce": false
    },
    "node": {
        "color": "#3498db",
        "fontColor": "#1e272e",
        "fontWeight": "normal",
        "highlightColor": "SAME",
        "highlightFontSize": 8,
        "highlightFontWeight": "normal",
        "highlightStrokeColor": "SAME",
        "highlightStrokeWidth": "SAME",
        "labelProperty": "id",
        "mouseCursor": "pointer",
        "opacity": 1,
        "renderLabel": true,
        "size": 200,
        "strokeColor": "#3498db",
        "strokeWidth": 1.5,
        "svg": "",
        "symbolType": "square",
        viewGenerator: node => <DependencyGraphNodeContainer data={node} />,
    },
    "link": {
        "color": "#2c3e50",
        "fontColor": "black",
        "fontSize": 8,
        "fontWeight": "normal",
        "highlightColor": "true",
        "highlightFontSize": 8,
        "highlightFontWeight": "normal",
        "labelProperty": "label",
        "mouseCursor": "pointer",
        "opacity": 1,
        "renderLabel": true,
        "semanticStrokeWidth": false,
        "strokeWidth": 1.5,
        "markerHeight": 6,
        "markerWidth": 6,
    }   
}




export const basicRelationshipData = {
    nodes: [
        {id: "object", group: 1},
        {id: "Class A", group: 1},
        {id: "Class B", group: 1},
        {id: "Class C", group: 1},
        {id: "Class D", group: 1},
        {id: "Class E", group: 1},
        {id: "Class F", group: 1},
    ],
    links: [
        {source: "object", target: "Class A", value: 1},
        {source: "Class A", target: "Class B", value: 1},
        {source: "Class A", target: "Class C", value: 1},
        {source: "Class B", target: "Class D", value: 1},
        {source: "Class D", target: "Class E", value: 1},
        {source: "Class E", target: "Class F", value: 1},
    ]
}

export const controlFlowGraphData = {
    nodes: [
        {id: "A = 0", group: 1},
        {id: "B > C", group: 1},
        {id: "A = B", group: 1},
        {id: "A = C", group: 1},
        {id: "end-if", group: 1},
        {id: "end-if-complete", group: 1},
    ],
    links: [
        {source: "A = 0", target: "B > C", value: 1},
        {source: "B > C", target: "A = B", value: 1},
        {source: "B > C", target: "A = C", value: 1},
        {source: "A = B", target: "end-if", value: 1},
        {source: "A = C", target: "end-if", value: 1},
        {source: "end-if", target: "end-if-complete", value: 1},
    ]
}

export const dependencyGraphData = {
    nodes: [
        {id: "preact.js", group: 1},
        {id: "render.js", group: 1},
        {id: "render-queue.js", group: 1},
        {id: "clone-element.js", group: 1},
        {id: "h.js", group: 1},
        {id: "vnode.js", group: 1},
        {id: "options.js", group: 1},
        {id: "util.js", group: 1},
        {id: "diff.js", group: 1},
        {id: "component.js", group: 1},
        {id: "index.js", group: 1},
        {id: "constants.js", group: 1},
    ],
    links: [
        {source: "preact.js", target: "render.js", value: 1},
        {source: "preact.js", target: "component.js", value: 1},
        {source: "preact.js", target: "clone-element.js", value: 1},
        {source: "preact.js", target: "h.js", value: 1},
        {source: "preact.js", target: "render-queue.js", value: 1},
        {source: "component.js", target: "util.js", value: 1},
        {source: "component.js", target: "options.js", value: 1},
        {source: "component.js", target: "constants.js", value: 1},
        {source: "render.js", target: "diff.js", value: 1},
        {source: "h.js", target: "options.js", value: 1},
        {source: "h.js", target: "vnode.js", value: 1},
        {source: "diff.js", target: "index.js", value: 1},
        {source: "index.js", target: "options.js", value: 1},
    ]
}

export const classDiagramData = {
    nodes: [
        {   id: "A",
            classArgs: {"Arg1": "String", "Arg2": "Int"}, 
            classFunctions: {
                "Function1": {
                    arguments: {
                        "Arg1": "String",
                        "Arg2": "Int",
                    },
                    returnType: "String"
            }}
        },
        {   id: "B", 
            classArgs: {"Arg1": "String", "Arg2": "Int"}, 
            classFunctions: {
                "Function1": {
                    arguments: {
                        "Arg1": "String",
                        "Arg2": "Int",
                    },
                    returnType: "String"
            }}
        }
    ],
    links: [
        {source: "A", target: "B", label: "test", value: 1},
 
    ]
}


export const configMapping = {
    "static": staticConfig,
    "dynamic": dynamicConfig,
    "classDiagram": classDiagramConfig,
    "dependencyGraph": dependencyGraphConfig,
}
