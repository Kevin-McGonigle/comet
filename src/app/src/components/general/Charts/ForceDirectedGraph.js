import React from 'react';
import { Graph } from 'react-d3-graph';
import { Pane } from 'evergreen-ui';
import { configMapping } from './configs';
import styles from './ForceDirectedGraph.css';
import classnames from 'classnames';

const cx = args => classnames(styles, args)

const ForceDirectedGraph = (props) => {
    const {
        data,
        graphType,
        title,
    } = props;

    const windowHeight = window.screen.height - 200;
    const windowWidth = window.screen.width - 350;
    const config = configMapping[graphType];
    config.height = windowHeight;
    config.width = windowWidth;

    return (
        <Pane
            display="flex"
            flexWrap="wrap"
            flexGrow={1}
            marginRight="10"
            background="tint2"
            elevation={4}   
        >
                        
            <div className={cx("title")}>{ title }<hr/></div>
            <Graph
                borderStyle="solid"
                id="graph-id"
                data={data}
                config={config}
            />
        </Pane>
    )
}

export default ForceDirectedGraph;
