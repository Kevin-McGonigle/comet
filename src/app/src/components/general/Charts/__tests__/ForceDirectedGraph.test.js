import React from 'react';
import ForceDirectedGraph from '../ForceDirectedGraph';
import { dependencyGraphData } from '../configs';
import { shallow } from 'enzyme';
import "../../../../setupTests"

test('should render ForceDirectedGraph as expected', () => { 
    const component = shallow(
        <ForceDirectedGraph 
            title="Dependency Graph"
            data={dependencyGraphData} 
            graphType="dynamic"         
        />
    );
    expect(component.html()).toMatchSnapshot();
});
