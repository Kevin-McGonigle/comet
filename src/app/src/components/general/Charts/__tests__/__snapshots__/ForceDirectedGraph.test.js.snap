// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ForceDirectedGraph as expected 1`] = `"<div class=\\"css-nil ðŸ“¦bs_kymczs ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-wrap_wrap ðŸ“¦flx-grow_1 ðŸ“¦mr_10 ðŸ“¦box-szg_border-box\\"><div class=\\"title\\">Dependency Graph<hr/></div><div id=\\"graph-id-graph-wrapper\\"><svg name=\\"svg-container-graph-id\\" style=\\"height:-200px;width:-350px\\"><defs><marker class=\\"marker\\" id=\\"marker-small\\" viewBox=\\"0 -5 10 10\\" refX=\\"16\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"#2c3e50\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-small-highlighted\\" viewBox=\\"0 -5 10 10\\" refX=\\"16\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"true\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-medium\\" viewBox=\\"0 -5 10 10\\" refX=\\"21.333333333333332\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"#2c3e50\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-medium-highlighted\\" viewBox=\\"0 -5 10 10\\" refX=\\"21.333333333333332\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"true\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-large\\" viewBox=\\"0 -5 10 10\\" refX=\\"26.666666666666664\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"#2c3e50\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-large-highlighted\\" viewBox=\\"0 -5 10 10\\" refX=\\"26.666666666666664\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"true\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker></defs><g id=\\"graph-id-graph-container-zoomable\\" style=\\"transition-duration:0s\\"><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"preact.js,render.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"preact.js,component.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"preact.js,clone-element.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"preact.js,h.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"preact.js,render-queue.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"component.js,util.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"component.js,options.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"component.js,constants.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"render.js,diff.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"h.js,options.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"h.js,vnode.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"diff.js,index.js\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"index.js,options.js\\"></path></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"preact.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">preact.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"render.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">render.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"render-queue.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">render-queue.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"clone-element.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">clone-element.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"h.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">h.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"vnode.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">vnode.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"options.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">options.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"util.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">util.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"diff.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">diff.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"component.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">component.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"index.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">index.js</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"constants.js\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,-7.978845608028654,0A7.978845608028654,7.978845608028654,0,1,1,7.978845608028654,0\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"11.5\\" dy=\\".35em\\" fill=\\"black\\" font-size=\\"8\\" font-weight=\\"normal\\" opacity=\\"1\\">constants.js</text></g></g></svg></div></div>"`;
